internal/tasks/core.go

```diff
@@ -80,15 +81,18 @@ func runNextTask() {
 		return
 	}
 
-	now := timestamp.Now()
-	if task.NextRunTime > now {
+	startTimestamp := timestamp.Now()
+	if task.NextRunTime > startTimestamp {
 		return
 	}
 
 	log.Debug("Task started.", task)
 	runTask(task)
+	endTimestamp := timestamp.Now()
```

Comment:
I wouldn't bother with this here.
You can just use `timestamp.Now()` in the call to stats.AsyncStoreCourseTaskDuration().
It would only be adding in the time of a log, which is pretty small.